Sources/Fuzzilli/Compiler/Compiler.swift

```diff
@@ -957,6 +994,18 @@ public class JavaScriptCompiler {
                 }
             }
 
+        case .callSuperConstructor(let callSuperConstructor):
+            let (arguments, spreads) = try compileCallArguments(callSuperConstructor.arguments)
+            let isSpreading = spreads.contains(true)
+            
+            if isSpreading {
+                throw CompilerError.unsupportedFeatureError("Spread arguments are not supported in super constructor calls")
+            }
+            guard !callSuperConstructor.isOptional else {
+                throw CompilerError.unsupportedFeatureError("Optional chaining is not supported in super constructor calls")
+            }
+            return emit(CallSuperConstructor(numArguments: arguments.count), withInputs: arguments).output
```

Comment:
Is this why you need `CallSuperConstructor` to now return a value? I wonder if we should just do this instead:
```
// The "return" value of invoking the super constructor is just |this|
emit(CallSuperConstructor(numArguments: arguments.count), withInputs: arguments)
return emit(LoadThis())
```
WDYT?

Reply:
Ah even simpler, I think you can just do `return lookupIdentifier("this");` since `this` must be available inside class constructors/methods (see https://github.com/googleprojectzero/fuzzilli/blob/0f521821d55f000acb5e63e9251274fc478ff456/Sources/Fuzzilli/Compiler/Compiler.swift#L226)